omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/PasswordResetController2_2.java

```diff
@@ -40,8 +42,17 @@ public class PasswordResetController2_2 extends BaseRestController {
 	@ResponseStatus(HttpStatus.OK)
 	public void requestPasswordReset(@RequestBody Map<String, String> body) throws MessageException {
 		String usernameOrEmail = body.get("usernameOrEmail");
-		User user = userService.getUserByUsernameOrEmail(usernameOrEmail);
-		if (user != null) {
+		Context.addProxyPrivilege(PrivilegeConstants.GET_USERS);
+		User user;
+		try {
+			user = userService.getUserByUsernameOrEmail(usernameOrEmail);
+		} finally {
+			Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
+		}
+
+		if (user == null || user.getUserId() == null ) {
```

Comment:
If the previous implementation did not throw an exception if user is null, we shouldn't do so here either.  For example, if you try to hit this, and get an exception, you know that username / email doesn't exist.  That may not be information we want to expose.

Just simplify this method and do everything inside of the try/finally block.  No need to declare the user outside of this block, and then use this user outside of this block.  Something like:

```
try {
  Context.addProxyPrivilege(PrivilegeConstants.GET_USERS);
  User user = userService.getUserByUsernameOrEmail(usernameOrEmail);
  if (user != null {
    userService.setUserActivationKey(user);
  }
}
finally {
  Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
}
```